Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/HttpResponse.java

```diff
@@ -21,21 +21,23 @@
  */
 package com.microsoft.azure.hdinsight.sdk.common;
 
-import java.util.HashMap;
+import org.apache.http.Header;
+
+import java.util.Arrays;
 import java.util.List;
-import java.util.Map;
+import java.util.stream.Collectors;
 
 public class HttpResponse {
     private int code;
     private String message;
-    private Map<String, List<String>> headers;
+    private List<Header> headers;
     private String content;
 
-    public HttpResponse(int code, String message,Map<String, List<String>> headers,
+    public HttpResponse(int code, String message, Header[] headers,
                         String content) {
         this.code = code;
         this.message = message;
-        this.headers = new HashMap<>(headers);
+        this.headers = Arrays.asList(headers);
```

Comment:
Will headers be null here? If it is, this code will hit NPE error.

Reply:
Fix it by adding NPE check here